webui/react/src/pages/F_ExpList/glide-table/contextMenu.tsx

```diff
@@ -0,0 +1,223 @@
+import { ExclamationCircleOutlined } from '@ant-design/icons';
+import { Menu, MenuProps } from 'antd';
+import { MenuInfo } from 'rc-menu/lib/interface';
+import React, { MutableRefObject, useCallback, useEffect, useRef } from 'react';
+
+import useModalExperimentMove from 'hooks/useModal/Experiment/useModalExperimentMove';
+import useModalHyperparameterSearch from 'hooks/useModal/HyperparameterSearch/useModalHyperparameterSearch';
+import usePermissions from 'hooks/usePermissions';
+import {
+  activateExperiment,
+  archiveExperiment,
+  cancelExperiment,
+  deleteExperiment,
+  killExperiment,
+  openOrCreateTensorBoard,
+  pauseExperiment,
+  unarchiveExperiment,
+} from 'services/api';
+import { ErrorLevel, ErrorType } from 'shared/utils/error';
+import { capitalize } from 'shared/utils/string';
+import { ExperimentAction as Action, ProjectExperiment } from 'types';
+import { modal } from 'utils/dialogApi';
+import handleError from 'utils/error';
+import { getActionsForExperiment } from 'utils/experiment';
+import { openCommandResponse } from 'utils/wait';
+
+const dropdownActions = [
+  // Action.SwitchPin, requires settings
+  Action.Activate,
+  Action.Pause,
+  Action.Archive,
+  Action.Unarchive,
+  Action.Cancel,
+  Action.Kill,
+  Action.Move,
+  Action.OpenTensorBoard,
+  Action.HyperparameterSearch,
+  Action.Delete,
+];
+
+// eslint-disable-next-line
+function useOutsideClickHandler(ref: MutableRefObject<any>, handler: () => void) {
+  useEffect(() => {
+    /**
+     * Alert if clicked on outside of element
+     */
+    function handleClickOutside(event: Event) {
+      if (ref.current && !ref.current.contains(event.target)) {
+        handler();
+      }
+    }
+    // Bind the event listener
+    document.addEventListener('mouseup', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mouseup', handleClickOutside);
+    };
+  }, [ref, handler]);
+}
+
+export interface TableContextMenuProps extends MenuProps {
+  fetchExperiments: () => void;
+  open: boolean;
+  experiment: ProjectExperiment;
+  handleClose: () => void;
+  x: number;
+  y: number;
+}
+
+export const TableContextMenu: React.FC<TableContextMenuProps> = ({
+  experiment,
+  fetchExperiments,
+  handleClose,
+  open,
+  x,
+  y,
+}) => {
+  const containerRef = useRef(null);
+  useOutsideClickHandler(containerRef, handleClose);
+
+  const permissions = usePermissions();
+
+  const onComplete = useCallback(() => {
+    handleClose();
```

Comment:
think it looks better to close the context menu before opening the modal, rather than after complete. included that as a part of [this](https://github.com/mapmeld/determined/pull/2)

Reply:
thanks 👍